api-module-library/hubspot/api.js

```diff
@@ -721,6 +743,29 @@ class Api extends OAuth2Requester {
         return this._patch(options);
     }
 
+    async pushSitePageDraftToLive(objId){
```

Comment:
I've been hesitant to press "play" on JSDoc for years. (Just searched Slack and as far back as 2018 we were discussing whether to implement.)

It seems like folks can overdo it, underdo it, and just have wildly varying ways of doing it.

If we can agree on a consistent, probably minimal style and knowing _when_ to document via JSDocs and when not, I'm game.

[Clean Code](https://www.amazon.com/Clean-Code-Handbook-Software-Craftsmanship/dp/0132350882) and that group is pretty opposed to in-line documentation... but I think they're mostly in the camp of "your code should self document, and you should have as minimal comment as possible"... @MichaelRyanWebber I think you were reading the book too?

Anywho, that's my weigh in. I'm game to adopt, if we can set a baseline consistent style that everyone agrees on/can easily adopt and adhere to.

Reply:
@seanspeaks - thanks for the input. Ok, let's not start a cosmic discussion about our code practices as part of this focused PR, then :) I am fine with keeping it as is.

I see some big fish like [Angular](https://github.com/angular/angular/blob/main/packages/core/src/render/api.ts) and [React](https://github.com/facebook/react/blob/c8deb5db660f3f6eeca30d29b4bd48d9e82b3792/packages/react/src/ReactElement.js#L128) using JSDoc in parts of their public API, which makes me wonder if that isn't useful.

Reply:
@seanspeaks I've more listened to talks by the author, and then discussions about his approach. 

I agree with the consistent and minimal approach, and personally would prefer options that:

- create a web presentation, even with examples a la the postman docs, sample data/response
- capture the more broad lessons of the underlying api which would help to communicate to the user, gotchas etc

and not having each function include a set of comments about it's inputs and outputs and side-effects. Especially with Typescript, I do hope this could be self-documenting, and if not, yeah probably the function is doing too much.